### PR TITLE
refactor: reset wallet service ignore flag

### PR DESCRIPTION
**Pull request by andreabadesso**
Solves https://github.com/HathorNetwork/hathor-wallet-mobile/issues/411

### Acceptance Criteria
- We should refactor the `LoadHistoryScreen` to be a function
- We should display our standard retry screen when the wallet-service fails
- If the error thrown from `wallet.start()` is not a request error, we should display the modal to send the error to our sentry
- We should have a mechanism to expire the ignore wallet service flag
- We should fix an issue preventing errors from being sent to Sentry

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/wallet.js [hunk 1]
```diff
@@ -194,11 +206,18 @@ export function* startWallet(action) {
     });
   } catch (e) {
     if (useWalletService) {
+      yield put(setShowWalletServiceLoadFailedModal(true));
+      yield take(types.WALLETSERVICE_RENDER_FAILED_ACK);
```
Comment (alexruzenhack) [inline src/sagas/wallet.js:210]: But if a user dismiss the modal without tapping on "ok" button? Should we consider it an acknowledge? Maybe the user wants to keep trying to use the wallet-service...

**Review by alexruzenhack — APPROVED** [review]

**Review by alexruzenhack — APPROVED** [review]

File: src/screens/LoadHistoryScreen.js [hunk 2]
```diff
@@ -21,66 +21,58 @@ import Spinner from '../components/Spinner';
...
+        fontSize: 18, lineHeight: 22, width: 200, textAlign: 'center'
+      }}
+      >
+        There&apos;s been an error connecting to the server
+      </Text>
+      <SimpleButton
+        containerStyle={{ marginTop: 12 }}
+        textStyle={{ fontSize: 18 }}
+        onPress={() => dispatch(onWalletReload())}
+        title='Try again'
```
Comment (pedroferreira1) [inline src/screens/LoadHistoryScreen.js:51]: Missing translation tag

File: src/sagas/wallet.js [hunk 3]
```diff
@@ -97,12 +100,22 @@ export function* isPushNotificationEnabled() {
  * @returns {Generator<unknown, boolean>}
  */
 export function* isWalletServiceEnabled() {
+  // Users might have had issues with the wallet-service in the past, we can detect
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:103]: If the `IGNORE_WS_TOGGLE_FLAG` changed, how can we detect old flags with this code?

File: src/sagas/wallet.js [hunk 4]
```diff
@@ -193,20 +206,28 @@ export function* startWallet(action) {
       password: pin,
     });
   } catch (e) {
+    // WalletRequestError can either be an network error making the request
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:209]:
```suggestion
    // WalletRequestError can either be a network error making the request
```

Comment (andreabadesso) [reply src/sagas/wallet.js]: Fixed, thanks

File: src/screens/LoadHistoryScreen.js (hunk 2 above)
Comment (andreabadesso) [reply src/screens/LoadHistoryScreen.js]: Fixed, thanks

File: src/sagas/wallet.js (hunk 3 above)
Comment (andreabadesso) [reply src/sagas/wallet.js:103]: You are right, I shouldn't change the flag, fixed in 2e37c2cccc323d9b6239d835e723d80fa26dd040

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**